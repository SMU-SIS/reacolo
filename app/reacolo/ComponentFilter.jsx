import React from 'react';

export default function ComponentFilter(props) {
  return props.rendered ? props.children : null;
}

ComponentFilter.propTypes = {
  rendered: React.PropTypes.bool.isRequired,
  children: React.PropTypes.element.isRequired
};
