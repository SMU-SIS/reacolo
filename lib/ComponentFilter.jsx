import React from 'react';

const ComponentFilter = (props) => {
  if (!props.rendered) {
    return null;
  } if (props.wrap) {
    return <div>{props.children}</div>;
  }
  return React.Children.only(props.children);
};

ComponentFilter.propTypes = {
  rendered: React.PropTypes.bool.isRequired,
  children: React.PropTypes.node.isRequired,
  wrap: React.PropTypes.bool,
};

ComponentFilter.defaultProps = {
  wrap: true
};


export default ComponentFilter;
