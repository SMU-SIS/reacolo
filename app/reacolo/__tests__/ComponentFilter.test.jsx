/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import ComponentFilter from '../ComponentFilter';

test('ComponentFilter renders itself and its children when rendered is true', () => {
  const component = renderer.create(
    <div id="container">
      <ComponentFilter rendered>
        <div id="child" />
      </ComponentFilter>
    </div>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ComponentFilter is not rendered when rendered is false', () => {
  const component = renderer.create(
    <div id="container">
      <ComponentFilter rendered={false}>
        <div id="child" />
      </ComponentFilter>
    </div>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
