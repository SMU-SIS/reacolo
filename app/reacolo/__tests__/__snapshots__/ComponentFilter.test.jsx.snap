exports[`test ComponentFilter is not rendered when rendered is false 1`] = `
<div
  id="container" />
`;

exports[`test ComponentFilter renders itself and its children when rendered is true 1`] = `
<div
  id="container">
  <div
    id="child" />
</div>
`;
